server/stream.go

```diff
@@ -3253,37 +3277,51 @@ func (mset *stream) internalLoop() {
 	mset.mu.RUnlock()
 
 	// Raw scratch buffer.
-	var _r [64 * 1024]byte
+	// This should be rarely used now so can be smaller.
+	var _r [1024]byte
 
 	for {
 		select {
 		case <-outq.ch:
 			pms := outq.pop()
 			for _, pmi := range pms {
 				pm := pmi.(*jsPubMsg)
-				c.pa.subject = []byte(pm.subj)
-				c.pa.deliver = []byte(pm.dsubj)
+				c.pa.subject = []byte(pm.dsubj)
```

Comment:
Again, this looks like a big difference, knowing that newJSPubMsg() still sets subj param to subj field. So unless all calls to newJSPubMsg() have had the passed param changed, wonder how this all worked?

Reply:
The embedded StoreMsg now has the origin subject, dsubj is what jsPubMsg needs to set c.pa.subject to. Looks weird since it got flipped but I believe is correct and consistent.

In newJSPubMsg I left the arg names the same but will flip to be consistent with struct naming.